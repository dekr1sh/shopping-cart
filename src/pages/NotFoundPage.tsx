import { Link } from "react-router-dom"
import { Button } from "@/components/ui/button"

export default function NotFoundPage() {
  return (
    <div className="container mx-auto flex min-h-[70vh] flex-col items-center justify-center px-4 text-center">
      <h2 className="text-3xl font-bold mb-4">404 - Page Not Found</h2>
      <p className="text-muted-foreground mb-8">Sorry, the page you're looking for doesn't exist.</p>
      <Button asChild> 
        <Link to="/">Go Home</Link>
      </Button>
    </div>
  )
}