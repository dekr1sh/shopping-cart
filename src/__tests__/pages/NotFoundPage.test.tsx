import { describe, it, expect } from "vitest";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { render } from "../test-utils";
import NotFoundPage from "@/pages/NotFoundPage";

describe("NotFoundPage", () => {
  it("matches the snapshot", () => {
    const { container } = render(<NotFoundPage />);
    expect(container).toMatchSnapshot();
  });

  it("navigates to home page when button is clicked", async () => {
    const user = userEvent.setup();

    render(<NotFoundPage />, { route: "/random-page" });
    const link = screen.getByRole("link", { name: /go home/i });

    await user.click(link);
    expect(window.location.pathname).toBe("/");
  })
});
