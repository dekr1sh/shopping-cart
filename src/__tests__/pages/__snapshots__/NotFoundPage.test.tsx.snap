// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFoundPage > matches the snapshot 1`] = `
<div>
  <div
    class="mx-auto flex min-h-[70vh] flex-col items-center justify-center px-4 text-center"
  >
    <h2
      class="text-3xl font-bold mb-4"
    >
      404 - Page Not Found
    </h2>
    <p
      class="text-muted-foreground mb-8"
    >
      Sorry, the page you're looking for doesn't exist.
    </p>
    <a
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3"
      data-discover="true"
      data-slot="button"
      href="/"
    >
      Go Home
    </a>
  </div>
     
</div>
`;
